allennlp/commands/elmo.py

```diff
@@ -0,0 +1,292 @@
+
+The input file is previously tokenized, whitespace separated text, one sentence per line.
+The output is a hdf5 file (<http://docs.h5py.org/en/latest/>) where each
+sentence is a size (3, num_tokens, 1024) array with the biLM representations.
+
+In the default setting, each sentence is keyed in the output file by the line number
+in the original text file.  Optionally, by specifying --use-sentence-key
+the first token in each sentence is assumed to be a unique sentence key
+used in the output file.
+
+#TODO(michaels) add a link to the ELMo paper once published.
+
+.. code-block:: bash
+
+   $ python -m allennlp.run elmo --help
+   usage: python -m allennlp.run [command] elmo [-h] [--vocab-path VOCAB_PATH]
+                                                [--options-file OPTIONS_FILE]
+                                                [--weight-file WEIGHT_FILE]
+                                                [--batch-size BATCH_SIZE]
+                                                [--cuda-device CUDA_DEVICE]
+                                                [--use-sentence-key]
+                                                input_file output_file
+
+   Create word vectors using ELMo.
+
+   positional arguments:
+     input_file            path to input file
+     output_file           path to output file
+
+   optional arguments:
+     -h, --help            show this help message and exit
+     --vocab-path VOCAB_PATH
+                           A path to a vocabulary file to generate
+     --options-file OPTIONS_FILE
+                           The path to the ELMo options file.
+     --weight-file WEIGHT_FILE
+                           The path to the ELMo weight file.
+     --batch-size BATCH_SIZE
+                           The batch size to use.
+     --cuda-device CUDA_DEVICE
+                           The cuda_device to run on.
+     --use-sentence-key
+"""
+
+import logging
+from typing import IO, List, Iterable, Tuple
+
+import argparse
+import h5py
+import torch
+
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.dataset import Batch
+from allennlp.data import Token, Vocabulary, Instance
+from allennlp.data.fields import TextField
+from allennlp.data.token_indexers.elmo_indexer import ELMoTokenCharactersIndexer
+from allennlp.nn.util import remove_sentence_boundaries
+from allennlp.modules.elmo import _ElmoBiLm
+from allennlp.commands.subcommand import Subcommand
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+DEFAULT_OPTIONS_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_options.json" # pylint: disable=line-too-long
+DEFAULT_WEIGHT_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_weights.hdf5" # pylint: disable=line-too-long
+DEFAULT_BATCH_SIZE = 64
+
+
+class Elmo(Subcommand):
```

Comment:
We already have an `Elmo` class in an `elmo.py` in `allennlp/modules/elmo.py`.  Can you rename this class (and potentially file) to something else to avoid confusion?

Reply:
I've found it annoying too to have two `elmo.py`s, but we seem to follow this example elsewhere and so I followed it.

```
$ find . -name bidaf.py                                                                                                                                                                                                                          
./allennlp/models/reading_comprehension/bidaf.py
./allennlp/service/predictors/bidaf.py
./build/lib/allennlp/models/reading_comprehension/bidaf.py
./build/lib/allennlp/service/predictors/bidaf.py
```